riscv/src/coprocessors.rs

```diff
@@ -204,10 +204,8 @@ fn poseidon_gl_call() -> String {
     let decoding = |i| {
         format!(
             r#"
-        addr <=X= {} + x10;
-        P{i} <== mload();
-        addr <=X= {} + x10;
-        tmp1 <== mload();
+        P{i}, tmp2 <== mload({} + x10);
```

Comment:
So if I understand this right, `tmp2` will always be 0, because the value `x10` is word-aligned and we're only adding multiples of 4 to it. Is that guaranteed? or should we assert that `tmp1` is 0 after the first mload?

Maybe you could add a comment on that.

Reply:
tmp2 is added here only because we can't leave it empty in the syntax, but before we didn't care what the alignment was

Reply:
but yes it's guaranteed with some assumptions:
- x10 is generated by LLVM as the address of the function argument, so it should be aligned
- the offsets are generated by the Rust closure below, so it's also 4-aligned.